docs/content/how-to/agent/create-heterogeneous-nodepools.md

```diff
@@ -0,0 +1,166 @@
+    --release-image=quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE}
+~~~
+
+## Create Heterogeneous Node Pool
+
+By default, previous section creates node pool on `x86_64` architecture, to create heterogeneous node pools, you can apply the below manifest with the architecture you want.
+
+~~~shell
+apiVersion: hypershift.openshift.io/v1beta1
+kind: NodePool
+metadata:
+  name: ${HOSTED_CLUSTER_NAME}
+  namespace: ${CLUSTERS_NAMESPACE}
+spec:
+  arch: ${ARCH_PPC64LE}
+  clusterName: ${HOSTED_CLUSTER_NAME}
+  management:
+    autoRepair: false
+    upgradeType: InPlace
+  nodeDrainTimeout: 0s
+  platform:
+    agent:
+      agentLabelSelector:
+        matchLabels:
+          inventory.agent-install.openshift.io/cpu-architecture: ${ARCH_PPC64LE}
+    type: Agent
+  release:
+    image: quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE}
+  replicas: 0
+~~~
+This will create node pool of architecture `ppc64le` with 0 replicas.
+
+~~~shell
+      agentLabelSelector:
+        matchLabels:
+          inventory.agent-install.openshift.io/cpu-architecture: ppc64le
+~~~
+This selector block is used to select the agents which are matching given label. Which will ensure it will select only the agents from `ppc64le` arch when it scaled.
+
+## Create Infraenv
+
+For heterogeneous node pools, need to create infraenv for each architecture you are going to have it in your HCP.
+i.e. Heterogeneous node pools with x86_64 and ppc64le architecture, need to create two infraenv with both architecture.
+
+~~~sh
+export SSH_PUB_KEY=$(cat $HOME/.ssh/id_rsa.pub)
+
+envsubst <<"EOF" | oc apply -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: ${HOSTED_CLUSTER_NAME}-${ARCH_X86}
+  namespace: ${HOSTED_CONTROL_PLANE_NAMESPACE}
+spec:
+  cpuArchitecture: ${ARCH_X86}
+  pullSecretRef:
+    name: pull-secret
+  sshAuthorizedKey: ${SSH_PUB_KEY}
+EOF
+
+envsubst <<"EOF" | oc apply -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: ${HOSTED_CLUSTER_NAME}-${ARCH_PPC64LE}
+  namespace: ${HOSTED_CONTROL_PLANE_NAMESPACE}
+spec:
+  cpuArchitecture: ${ARCH_PPC64LE}
+  pullSecretRef:
+    name: pull-secret
+  sshAuthorizedKey: ${SSH_PUB_KEY}
+EOF
+~~~
+This will create two infraenv with x86_64 & ppc64le architectures. Before creating this, need to ensure respective architecture's OS image is added in `AgentServiceConfig`
+
+After this you can use the above infraenvs to get the minimal iso and follow the [create-agent-cluster.md](create-agent-cluster.md) to add them to your cluster as agents.
+
+## Scale the NodePool
+
+Once your agents are approved, you can scale the node pools. `agentLabelSelector` configured in node pool ensures that only matching agents gets added to the cluster.
+This will help in descaling the node pool as well, when you want to remove a specific arch nodes from the cluster, you can descale the respective node pool.
```

Comment:
```suggestion
This will help in descaling the nodepool as well, when you want to remove a specific arch nodes from the cluster, you can descale the respective nodepool.
```